### PR TITLE
Feature/74 implement and integrate checkbox component

### PR DESCRIPTION
**Pull request by SimerdeepSinghGrewal**
I have created a new Checkbox component to accept the following props:
label: to specify the text label associated with the checkbox.
checked: to determine whether the checkbox is initially checked.
onChange: to handle the change event when the checkbox is clicked.
variant: to define the checkbox style (e.g., primary, secondary).
className: to apply additional CSS classes.
style: to apply inline styles.
the component can handle different types of checkboxes, such as primary, secondary, etc.
I have tested everything as well, check screenshot
Screenshot 2024-06-20 162007

### CODE REVIEW

File: frontend/src/components/Checkbox/Checkbox.jsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    newChecked[index] = event.target.checked;
+    setChildChecked(newChecked);
+  };
+
+  const isIndeterminate = childChecked.some((child) => child !== childChecked[0]);
+
+  const getCheckboxStyle = () => {
+    if (variant === 'primary') {
+      return {
+        color: 'var(--main-purple)',
```
Comment (erenfn) [inline frontend/src/components/Checkbox/Checkbox.jsx:37]: Move the CSS inside the css file. Just return the css class in this funtcion.

**Review by erenfn — CHANGES_REQUESTED** [review]

File: frontend/src/components/Checkbox/Checkbox.jsx (hunk 1 above)
Comment (erenfn) [reply frontend/src/components/Checkbox/Checkbox.jsx:37]: Then it should be good to go

**Comment by erenfn** [comment]
Can you also add this to your code as a block comment?

```
label: to specify the text label associated with the checkbox.
checked: to determine whether the checkbox is initially checked.
onChange: to handle the change event when the checkbox is clicked.
variant: to define the checkbox style (e.g., primary, secondary).
className: to apply additional CSS classes.
style: to apply inline styles.
```

File: frontend/src/components/Checkbox/Checkbox.jsx (hunk 1 above)
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/Checkbox/Checkbox.jsx]: Done

**Comment by SimerdeepSinghGrewal** [comment]
> Can you also add this to your code as a block comment?
> 
> ```
> label: to specify the text label associated with the checkbox.
> checked: to determine whether the checkbox is initially checked.
> onChange: to handle the change event when the checkbox is clicked.
> variant: to define the checkbox style (e.g., primary, secondary).
> className: to apply additional CSS classes.
> style: to apply inline styles.
> ```

Thank you for this, I was thinking to add comments but then forgot.

**Review by erenfn — APPROVED** [review]
I assume nothing has changed in terms of functionality. (You can still create the checkboxes below) If so, you can merge it.
image

**Pull request merged**